lib/confluence/set_ops.es6.js

```diff
@@ -252,9 +252,12 @@ foam.CLASS({
 
   listeners: [
     function onFind(data, o) {
-      if (data.errored === true || data.found !== null) return;
+      // Return if already reject()ed or resolve()d promise.
+      if (data.errored === true || data.found) return;
+      // Resolve with value of "o" if null previously found, or o is non-null.
+      var shouldResolve = data.found === null || o !== null;
       data.found = o;
-      if (o !== null) data.resolve(data.found);
+      if (shouldResolve) data.resolve(data.found = o);
```

Comment:
Why the |= o| here? Did you mean to delete the previous line so that the assignment was only done when resolving the promise?